### PR TITLE
Added field to add ERC20 for the stage

### LINKED ISSUE #47: Need a button for the owner to commit some of his ERC20 token.
Do this in the create stage page. Let the owner choose how much erc20 tokens they want to send to the stage contract.

### PR DESCRIPTION
**Pull request by UsamaSaleem14**
In order to send eth to a contract the contract should have a receive or a payable function. We do have a payable function but it is being used for something else. @TBomberman , please add a receive function to send the ethers to the contract  by the owner.

### CODE REVIEW

File: scripts/deploy.ts [hunk 1]
```diff
@@ -81,6 +81,8 @@ export async function main() {
       await txStage.wait();
     }
   }
+  const testERC20 = await TestERC20.deploy("Test ERC20", "TestERC20");
```
Comment (TBomberman) [inline scripts/deploy.ts]: Isn't this already done on line 56-57?

File: src/components/Stages/AddStage/index.tsx [hunk 2]
```diff
@@ -170,6 +171,25 @@ function AddStage() {
                   })}
               </div>
 
+              <div className={`${style.inputDiv} ${style.targetInputMargin}`}>
```
Comment (TBomberman) [inline src/components/Stages/AddStage/index.tsx]: The erc20 token is already added when he created the campaign. That is what is expected to be sent to the stage.

**Review by TBomberman — COMMENTED** [review]
We just need a button so that the campaign erc20 token can be sent to the stage by the owner.

File: src/components/Stages/AddStage/index.tsx (hunk 2 above)
Comment (UsamaSaleem14) [reply src/components/Stages/AddStage/index.tsx]: Where are we adding the amount when we create the campaign?

File: scripts/deploy.ts (hunk 1 above)
Comment (UsamaSaleem14) [reply scripts/deploy.ts]: Fixed

File: src/components/Stages/AddStage/index.tsx (hunk 2 above)
Comment (TBomberman) [reply src/components/Stages/AddStage/index.tsx]: We haven't deposited any amount yet, this task is to add a button for it to be added after the campaign is created.

Comment (UsamaSaleem14) [reply src/components/Stages/AddStage/index.tsx]: Done. Added a modal in single stage page to commit erc

File: src/components/Stages/CommitERCTokenModal.tsx [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+type ConnectorModalProps = {
+  isOpen: boolean;
+  closeModal: (value?: boolean) => void;
+  commitERCAmount: () => Promise<void>;
+  setERCAmount: Dispatch<SetStateAction<number>>;
+  ercBtnDisable: boolean;
+};
+
+function CommitERCTokenModal({
```
Comment (TBomberman) [inline src/components/Stages/CommitERCTokenModal.tsx:12]: Can rename this to CommitTokenModal?

Comment (TBomberman) [reply src/components/Stages/CommitERCTokenModal.tsx:12]: Also there needs to be a CommitTokenModalHandler

File: src/hooks/useSingleStageHandler.ts [hunk 4]
```diff
@@ -621,6 +679,56 @@ export const useSingleStageHandler = () => {
...
+    if (data) {
+      setErcBtnDisable(true);
+      const erc20Contract = await getSmartContractWithSigner(
+        data.tokenAddress,
+        signer,
+        JSON.stringify(IERC20.abi)
+      );
+      const decimals = await erc20Contract.decimals();
+      const amount = ethers.utils.parseUnits(ercAmount.toString(), decimals);
+      await erc20Contract.transfer(stageAddress, amount);
```
Comment (TBomberman) [inline src/hooks/useSingleStageHandler.ts]: Does this need a try catch?

**Review by TBomberman — APPROVED** [review]

File: src/hooks/useSingleStageHandler.ts (hunk 4 above)
Comment (UsamaSaleem14) [reply src/hooks/useSingleStageHandler.ts]: Done

**Pull request merged**